RedHatInsights/tasks-frontend
# feat(RHINENG-3781): Include report and report_json in the extended CSV/JSON report

Implements https://issues.redhat.com/browse/RHINENG-3781.

This improves the CSV and JSON export of the completed task report. In addition to System name, Status, Message columns, it now also includes further information about the executed job (from `report`, `report_json` fields).

- The `useExport` hook now supports `prepareItems` that allows to pre-process the items from the table before the CSV report is generated and `extraExportColumns` that extend the visible columns with the columns that will be added in the final CSV/JSON export.

## How to test

Go to /tasks/executed and find any executed job. Try to find the results with some findings (warnings, errors, info messages) or just with some plain report. Export the table as CSV/JSON structure and verify it's correct and up to the information present in the UI.
